src/Cryptol/Utils/Ident.hs

```diff
@@ -394,10 +397,15 @@ instance NFData MaybeAnon
 maybeAnonText :: MaybeAnon -> Text -> Text
 maybeAnonText mb txt =
   case mb of
-    NormalName       -> txt
-    AnonModArgName   -> "`where` argument of " <> txt
-    AnonIfaceModName -> "`parameter` interface of " <> txt
-    AnonInstImport   -> txt
+    NormalName -> txt
+    AnonModArgName l c
+      | T.null txt -> "where_at__" <> suff l c
+      | otherwise  -> txt <> "__where"
+    AnonIfaceModName    -> txt <> "__parameter"
+    AnonInstImport l c  -> "import_at__" <> suff l c
```

Comment:
Worth leaving a comment here for future developers to remind them to update the comments in the reference manual if this code is ever changed.